enhancements/improve-kubeletconfig-default-configuration-check-enhancement.md

```diff
@@ -0,0 +1,278 @@
+  "podPidsLimit": 4096,
+  "resolvConf": "/etc/resolv.conf",
+  "cpuCFSQuota": true,
+  "cpuCFSQuotaPeriod": "100ms",
+  "nodeStatusMaxImages": 50,
+  "maxOpenFiles": 1000000,
+  "contentType": "application/vnd.kubernetes.protobuf",
+  "kubeAPIQPS": 50,
+  "kubeAPIBurst": 100,
+  "serializeImagePulls": false,
+  "evictionHard": {
+    "imagefs.available": "15%",
+    "memory.available": "100Mi",
+    "nodefs.available": "10%",
+    "nodefs.inodesFree": "5%"
+  },
+  "evictionPressureTransitionPeriod": "5m0s",
+  "enableControllerAttachDetach": true,
+  "makeIPTablesUtilChains": true,
+  "iptablesMasqueradeBit": 14,
+  "iptablesDropBit": 15,
+  "featureGates": {
+    "APIPriorityAndFairness": true,
+    "CSIMigrationAWS": false,
+    "CSIMigrationAzureFile": false,
+    "CSIMigrationGCE": false,
+    "CSIMigrationvSphere": false,
+    "DownwardAPIHugePages": true,
+    "PodSecurity": true,
+    "RotateKubeletServerCertificate": true
+  },
+  "failSwapOn": true,
+  "memorySwap": {},
+  "containerLogMaxSize": "50Mi",
+  "containerLogMaxFiles": 5,
+  "configMapAndSecretChangeDetectionStrategy": "Watch",
+  "systemReserved": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "enforceNodeAllocatable": [
+    "pods"
+  ],
+  "volumePluginDir": "/etc/kubernetes/kubelet-plugins/volume/exec",
+  "logging": {
+    "format": "text",
+    "flushFrequency": 5000000000,
+    "verbosity": 2,
+    "options": {
+      "json": {
+        "infoBufferSize": "0"
+      }
+    }
+  },
+  "enableSystemLogHandler": true,
+  "shutdownGracePeriod": "0s",
+  "shutdownGracePeriodCriticalPods": "0s",
+  "enableProfilingHandler": true,
+  "enableDebugFlagsHandler": true,
+  "seccompDefault": false,
+  "registerNode": true,
+  "kind": "KubeletConfiguration",
+  "apiVersion": "kubelet.config.k8s.io/v1beta1"
+}
+```
+
+When we get `KubeletConfig` for each node, we will compare them, and
+only save the consistent result to a file at `/api/v1/nodes/proxy/configz`.
+
+In the CaC content, we just need to check one location `/api/v1/nodes/proxy/configz`
+for `KubeletConfig` default configuration.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters with many nodes, I'd like to run
+  compliance scans on my fleet in such a way that we have less compliance rule failures when
+  no reboot remediation has been applied.
+
+### API Extensions
+
+None is changed
+
+
+### Risks and Mitigations
+
+There are no risks or mitigations for this change.
+
+## Design Details
+
+### Open Questions [optional]
+
+
+1. Is `/api/v1/nodes/{NODE_NAME}/proxy/configz` a good place to check
+   default `KubeletConfig`?
+
+   Yes, it is a good place to check default `KubeletConfig`
+
+2. If we want to check the KubeletConfig through Kubernetes API, the rule
+   needs to change from node-rule to platform rule. Since platform rules
+   does not care about node type, it will require all nodes to have the 
+   same `KubeletConfig` configuration, wondering if that is a good idea?
```

Comment:
I think the question to ask is: is this possible and supported? Because if it is, someone is doing that and will ask us to support their deployment.

Reply:
I don't think we support cluster where the same pool has different KubeletConfig, because they should all be configured through KubeletConfig CRs, and it applies to a pool instead of individual node.

Reply:
As long as we have different checks for different pools, I'm fine.